### PR TITLE
solidity grammar: add a new rule `usingAliases` for `usingDirective`

### PR DESCRIPTION
**Pull request by jeasonstudio**
Thank you very much for providing the perfect solidity antlr4 grammar in the official repository, I encountered some problems when using it to build the [solidity-antlr4](https://github.com/jeasonstudio/solidity-antlr4) project (ast parser). This Pull-Request fixes the issue.

In the original grammar definition, `usingDirective` is hard to parse by the parser runtime:
https://github.com/ethereum/solidity/blob/ec563a12cb9247d985df1e52cfacd9d03520fa2c/docs/grammar/SolidityParser.g4#L336-L340

For example, my code is like this:
```solidity
using { A, B as + } for uint;
```

Here is how my parser handles it (using antlr4 typescript as an example):
```ts
const visitUsingDirective = (ctx: UsingDirectiveContext) => {
  const isUsingAlias = !!ctx.LBrace() && !!ctx.RBrace(); // not good.
  if (!isUsingAlias) {
    const name = ctx.identifierPath(0); // shouldn't be an array.
  } else {
    const names = ctx.identifierPath(); // symbols.
    const alias = ctx.userDefinableOperator(); // aliases(operators), but it can't correspond with symbol one by one.
  }
};
```

So I added a new rule `usingAliases` to solve this problem, and the rule designed refer to the definition of `importDirective`:
```g4
usingDirective:
  Using (
    identifierPath
    | (LBrace usingAliases (Comma usingAliases)* RBrace)
  ) For (Mul | typeName) Global? Semicolon;

usingAliases: identifierPath (As userDefinableOperator)?;
```

Hope it can be processed soon, thanks.

### CODE REVIEW

**Comment by jeasonstudio** [comment]
There is an unrelated error in Circle CI. How can I rerun the test in ci? @cameel

**Review by r0qs — COMMENTED** [review]
Hi @jeasonstudio thanks for the contribution. LGTM, could you please rebase it?

**Comment by jeasonstudio** [comment]
> Hi @jeasonstudio thanks for the contribution. LGTM, could you please rebase it?

I have rebased this branch, but ci reported an error, and I don‘t think it has anything to do with my pr.

Thank you for your review. @r0qs

**Review by r0qs — APPROVED** [review]
Thanks! Yes, it is indeed unrelated (I did a workaround here: https://github.com/ethereum/solidity/pull/14833). So I'm approving this anyway. But you may need to rebase it again if we merge the other PR first ;)

**Comment by jeasonstudio** [comment]
> Thanks! Yes, it is indeed unrelated (I did a workaround here: #14833). So I'm approving this anyway. But you may need to rebase it again if we merge the other PR first ;)

done
